src/code/entities/hard_hat_beetle.asm

```diff
@@ -1,3 +1,7 @@
+; Note: this entity, unlike most others, use 3 spriteslots (out of 4 available).
```

Comment:
The hardhat beetle isn't the enemy that has this property, it actually fits in a 2 slots.
The giant buzz blob has the property of using 3 slots as well as the stone hinox.

Other enemies that use more slots are:
- D4 boss (fish, always forget the name) but that can load everything due to a warp
- Evil Eagle, once again, loads everything on warp
- Nightmare, loads 4 sets on warp in, but switches out sprites even during battle (didn't check but I assume it uses the interrupt based mechanism)
- Slime boss (D3) uses 2 sets for the boss, but also 1 for the small slimes, so has 3 as well, but this one has the small slimes "preloaded" so it only needs to load 2 when entering the boss room.

There are a bunch of NPCs that have more then 2 spritesheets required as well, but those are all indoor, and thus behind warps (sun fish and manu are two examples)
The sneaky one here is the walrus, which uses 2 spritesheets, but then another one for Marin singing, so 3. It uses the same trick as the slime boss, so the walrus sprites are loaded 1 room before the actual walrus. So Marin's singing sprite can be loaded when entering the room with the walrus.